### DISCUSSION TITLE
Is this a viable solution to finding all types of Flat Config?

### DISCUSSION THREAD

**Discussion by will-stone** [eslint/eslint#17574] [Ideas] [answered]
Hi,

I've been converting my projects over to the flat config. Kudos on that, by the way, it's really quite fun to use; I've [now got a config which configures itself depending on the environment it's in](https://github.com/will-stone/dx/blob/main/packages/eslint-config/eslint-config.js) 🤓 What I've found is that I have to use `ESLINT_USE_FLAT_CONFIG=true eslint -c eslint.config.mjs .` in non-ESM environments. I am wondering, before I spend any time looking into ESLint internals, writing tests etc, is this a viable solution to auto-finding all flat config extensions (albeit with these values in constants)?

![CleanShot 2023-09-17 at 9 26 07@2x](https://github.com/eslint/eslint/assets/654103/2b5f5a12-d0ec-45c9-91f5-d73b85dd8d6d)

https://github.com/eslint/eslint/blob/main/lib/eslint/flat-eslint.js#L248

Thanks,

Will.

**Comment by nzakas** [comment, marked as answer]
I'm not quite sure what you're asking here. Are you asking if a PR with this change would be accepted? If that's the question, the answer is no, we don't want to support multiple different file extensions for the config file. We are rapidly moving to a world where ESM is the only format people will be using, so it doesn't make sense to add extra affordances for CommonJS.

If you're asking would this change work in theory, then yes, it would.

If you're asking something else, please clarify.

**Reply by will-stone** [reply]
Thanks @nzakas, to be honest you’ve covered all bases 😅 I did give it a go and couldn’t get it to work but good to know it’s not worth me spending any more time on it. And I agree, let’s ESM all the things if we can 👍
